File: Models/ViewModels/ReportVM.cs
using GoWheels_WebAPI.Models.Entities;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class ReportVM : BaseModelVM
    {
        public int Id { get; set; }
        public string? Content { get; set; }
        public required int PostId { get; set; }
        public PostVM Post { get; set; } = null!;
        public required int ReportId { get; set; }
        public string? ReportName { get; set; }
    }
}

File: FE/src/Model/ReportVM.js
// src/models/ReportVM.js

import PostVM from './PostVM';

export default class ReportVM {
  constructor(data = {}) {
    this.id = data.id || 0;
    this.content = data.content || '';
    this.postId = data.postId || 0;
    this.post = new PostVM(data.post || {});
    this.reportId = data.reportId || 0;
    this.reportName = data.reportName || '';
  }
}
